core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -333,7 +333,8 @@ private Object handleNonTxWriteCommand(InvocationContext ctx, WriteCommand comma
 
          // We increment the topology id so that updateTopologyIdAndWaitForTransactionData waits for the next topology.
          // Without this, we could retry the command too fast and we could get the OutdatedTopologyException again.
-         if (trace) log.tracef("Retrying command because of topology change, current topology is %d: %s", command);
+         if (trace) log.tracef("Retrying command because of topology change, current topology is %d: %s",
+                 command.getTopologyId(), command);
```

Comment:
The topology id is supposed to be from the current topology, not from the command. Something like this:

         int currentTopologyId = currentTopologyId();
         if (trace) log.tracef("Retrying command because of topology change, current topology is %d: %s",
               currentTopologyId, command);
         int newTopologyId = Math.max(currentTopologyId, command.getTopologyId() + 1);